src/libs/actions/Welcome.ts

```diff
@@ -98,47 +111,46 @@ Onyx.connect({
-        const isWorkspaceRoute = topRoute.name === 'Settings' && topRoute.params.path.includes('workspace');
-        const transitionRoute = _.find(routes, (route) => route.name === SCREENS.TRANSITION_BETWEEN_APPS);
-        const exitingToWorkspaceRoute = lodashGet(transitionRoute, 'params.exitTo', '') === 'workspace/new';
-        const openOnAdminRoom = lodashGet(topRoute, 'params.openOnAdminRoom', false);
-        const isDisplayingWorkspaceRoute = isWorkspaceRoute || exitingToWorkspaceRoute;
+        const topRoute = routes.length > 0 ? routes[routes.length - 1] : undefined;
+        const isWorkspaceRoute = topRoute !== undefined && topRoute.name === 'Settings' && topRoute.params?.path.includes('workspace');
+        const transitionRoute = routes.find((route) => route.name === SCREENS.TRANSITION_BETWEEN_APPS);
+        const exitingToWorkspaceRoute = transitionRoute?.params?.exitTo === 'workspace/new';
+        const openOnAdminRoom = topRoute === undefined ? undefined : topRoute.params?.openOnAdminRoom ?? false;
+        const isDisplayingWorkspaceRoute = isWorkspaceRoute ?? exitingToWorkspaceRoute;
 
         // If we already opened the workspace settings or want the admin room to stay open, do not
         // navigate away to the workspace chat report
         const shouldNavigateToWorkspaceChat = !isDisplayingWorkspaceRoute && !openOnAdminRoom;
 
-        const workspaceChatReport = _.find(
-            allReports,
-            (report) => ReportUtils.isPolicyExpenseChat(report) && report.ownerAccountID === currentUserAccountID && report.statusNum !== CONST.REPORT.STATUS.CLOSED,
-        );
+        const workspaceChatReport = Object.values(allReports ?? {}).find((report) => {
+            if (report) {
+                return ReportUtils.isPolicyExpenseChat(report) && report.ownerAccountID === currentUserAccountID && report.statusNum !== CONST.REPORT.STATUS.CLOSED;
+            }
+            return false;
+        });
 
-        if (workspaceChatReport || openOnAdminRoom) {
+        if (workspaceChatReport ?? openOnAdminRoom) {
             // This key is only updated when we call ReconnectApp, setting it to false now allows the user to navigate normally instead of always redirecting to the workspace chat
             Onyx.set(ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER, false);
         }
 
         if (shouldNavigateToWorkspaceChat && workspaceChatReport) {
-            Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(workspaceChatReport.reportID));
+            if (workspaceChatReport.reportID !== null) {
+                Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(workspaceChatReport.reportID));
+            }
 
             // If showPopoverMenu exists and returns true then it opened the Popover Menu successfully, and we can update isFirstTimeNewExpensifyUser
             // so the Welcome logic doesn't run again
-            if (showPopoverMenu()) {
+            if (showPopoverMenu?.()) {
```

Comment:
```suggestion
            if (showPopoverMenu ? showPopoverMenu() : false) {
```

Reply:
I just wanted the code to be a little bit more explicit. but we can keep the original change if we prioritize concision